### ISSUE TITLE
FEDERATED SERVICE query fails on VALUES assignment

### ISSUE THREAD

**Issue by milung** [eclipse-rdf4j/rdf4j#5218]
### Current Behavior

When executing following query :

```sparql
SELECT * WHERE {
SERVICE <http://localhost:8080/rdf4j-server/repositories/test> {
    VALUES   ?o   { "a" } 
    ?s a ?o.
} }
```

I got response : 

```plain
Query evaluation error: org.eclipse.rdf4j.query.QueryEvaluationException: Encountered " <STRING_LITERAL2> "\"a\" "" at line 1, column 388. Was expecting one of: "(" ... <NIL> ... <VAR1> ... <VAR2> ...
```
This is the most simple materialization of the error originally found in a more complex queries involving graph patterns and subqueries. 

The error is not materialized if 
* variable ?o is not present in the subsequent triple, e.g. renaming it to o2
*  non federate query - if SERVICE is not used,
* using BIND( "a" AS ?o) instead of VALUES clause. 

The error materializes: 
* if subquery in SERVICE is used, 
* with var value being IRI or literal
* with subject or object variable used
* with subgroups usage (graph or simple)

### Expected Behavior

there shall be no difference between usage of query through FEDERATED query or directly on the sparql endpoint

### Steps To Reproduce

1. Run dockerized rdf4j-workbench as per instructions at https://hub.docker.com/r/eclipse/rdf4j-workbench (tag: 5.1.0)
2.Open the site http://localhost:8080/rdf4j-workbench
3. Create new repository test (Native Store, no changed configurations) 
4. go to Query page and executed query: 

   ```sparql
   SELECT * WHERE {
   SERVICE <http://localhost:8080/rdf4j-server/repositories/test> {
     VALUES   ?o   { "a" } 
     ?s a ?o.
     } }
   ```
5. See error: 

   ```plain
   Query evaluation error: org.eclipse.rdf4j.query.QueryEvaluationException: Encountered " <STRING_LITERAL2> "\"a\" "" at line 1,  column 388. Was expecting one of: "(" ... <NIL> ... <VAR1> ... <VAR2> ...
   ```
6. Execute query (non federated)
   ```sparql
   SELECT * WHERE {
        VALUES   ?o   { "a" } 
     ?s a ?o.
     }
   ```
7. You get empty result set as expected. 

### Version

5.1.0

### Are you interested in contributing a solution yourself?

Perhaps?

### Anything else?

May be also related to this issue: #4522, it seems to be a result of sending optimized/simplified query to a federated service. An ability to disable optimization for federated queries will be at least a good workaround for such issues.

**Comment by hmottestad** [comment]
Does it happen if you use the full syntax?

`VALUES  ( ?o )   { ( "a" ) } `

**Comment by milung** [comment]
> Does it happen if you use the full syntax?
> 
> `VALUES ( ?o ) { ( "a" ) } `

Yes, I tried variaous forms, including multiple variables form. Except of BIND syntax it always failed

**Comment by aschwarte10** [comment]
This looks like the same issue as was resolved in https://github.com/eclipse-rdf4j/rdf4j/issues/5358 a while ago. I propose to close. Please recheck in your environment with a recent RDF4J version and reopen if problems still exist.
